components/google-cloud/google_cloud_pipeline_components/aiplatform/utils.py

```diff
@@ -575,13 +575,26 @@ def component_yaml_generator(**kwargs):
             component_param_name = component_param_name_to_mb_sdk_param_name.get(
                 key, key
             )
+            component_param_type = None
             if isinstance(value,
                           kfp.dsl._pipeline_param.PipelineParam) or serializer:
                 if is_mb_sdk_resource_noun_type(param_type):
                     metadata_type = map_resource_to_metadata_type(param_type)[1]
                     component_param_type = metadata_type
                 else:
-                    component_param_type = 'String'
+                    if param_type == int:
+                        component_param_type = 'Integer'
+                    elif param_type == float:
+                        component_param_type = 'Float'
+                    elif param_type == bool:
+                        component_param_type = 'Bool'
+                    elif param_type == list:
```

Comment:
Annotation checking for `list` and `dict` won't always work. The Vertex SDK generally uses typing.List and typing.Dict to allow further type qualification.

This method is a more exhaustive check that should work with the type annotations in the Vertex SDK: https://github.com/kubeflow/pipelines/blob/dec03067ca1f89f1ca23c7397830d60201448fa6/components/google-cloud/google_cloud_pipeline_components/aiplatform/utils.py#L131

It should be split into seperate checks because it's currently used to select the serializer (`json.dumps` work for both).

Something like:

```Python
def is_annotation_sequence(annotation: Any) -> bool:
    """Checks if the type is serializable.

    Args:
        annotation: parameter annotation
    Returns:
        True if serializable to json.
    """
    serializable_types = (list, collections.abc.Sequence, Sequence)
    return getattr(annotation, '__origin__', None) in serializable_types


def is_annotation_dict(annotation: Any) -> bool:
    """Checks if the type is serializable.

    Args:
        annotation: parameter annotation
    Returns:
        True if serializable to json.
    """
    serializable_types = (dict, Dict)
    return getattr(annotation, '__origin__', None) in serializable_types

def is_serializable_to_json(annotation: Any) -> bool:
    """Checks if the type is serializable.

    Args:
        annotation: parameter annotation
    Returns:
        True if serializable to json.
    """
    return is_annotation_sequence(annotation) or is_annotation_dict(annotation)
```

```Python
elif is_annotation_sequence(annotation):
  component_param_type = 'List'
elif is_annotation_dict(annotation):
  component_param_type = 'Dict'
...
```

Also this is not currently handling Proto parameters.

If this is just to unblock `int` then  one of the above are blockers.

Reply:
Thank you Sasha great point! updated the code accordingly. I moved the list and dict variations inline to simplify the logic since  the is_json_serializable already supports them. Also we recently added support for proto types see [here](https://github.com/kubeflow/pipelines/blob/808712381b173d2ba3753d9ead4b7bd7c75f9164/components/google-cloud/google_cloud_pipeline_components/aiplatform/utils.py#L176_ ) though good call I will add they to types as well. However since the expectation is that proto types are passed as serialized String I'll map them to String.
 
```python 
                    elif param_type in (list, collections.abc.Sequence, Sequence):
                        component_param_type = 'List'
                    elif param_type in (dict, Dict):
                        component_param_type = 'Dict'
                    elif param_type in PROTO_PLUS_CLASS_TYPES:
                        component_param_type = 'String'
                    else:
                        # For ProtoPlus and str set type to str
                        component_param_type = 'String'
```